### PR TITLE
allow custom source folders in buildozer.spec

### PR DESCRIPTION
**Pull request by kived**
Allows you to specify custom folders to use when building recipes:

    requirements.source.kivy = ../../kivy

This will set `P4A_kivy_DIR` for the Android target and `kivy_DIR` for the iOS target. The iOS portion is completely untested, as I don't have a Mac or an iOS device.

### CODE REVIEW

**Comment by tito** [comment]
Can you rebase with the latest master? That would be a good addition :)
(the ios was still based on the previous toolchain, not working anymore)

**Comment by kived** [comment]
Rebase complete :)

File: buildozer/targets/ios.py [hunk 1]
```diff
@@ -128,6 +128,17 @@ def compile_platform(self):
         if last_requirements != ios_requirements:
             need_compile = 1
 
+        source_dirs = {'{}_DIR'.format(name[20:]):
```
Comment (tito) [inline buildozer/targets/ios.py]: the ios toolchain require the name to be upper()

File: buildozer/targets/android.py [hunk 2]
```diff
@@ -464,6 +464,17 @@ def compile_platform(self):
         if not exists(dist_dir):
             need_compile = 1
 
+        source_dirs = {'P4A_{}_DIR'.format(name[20:]):
```
Comment (tito) [inline buildozer/targets/android.py]: Where the 20 came from?

**Comment by kived** [comment]
Alright, I added `.upper()` for iOS and added comments to explain the `[20:]` part.

**Pull request merged**